extensions/amp-story/0.1/amp-story-auto-ads.js

```diff
@@ -29,42 +34,151 @@ export class AmpStoryAutoAds extends AMP.BaseElement {
   constructor(element) {
     super(element);
 
-    /** @private {?AmpStory} */
+    /** @private {?./amp-story.AmpStory} */
     this.ampStory_ = null;
+
+    /** @private {?./navigation-state.NavigationState} */
+    this.navigationState_ = null;
+
+    /** @private {number} */
+    this.interactions_ = 0;
+
+    /** @private {!Array} */
+    this.adElements_ = [];
+
+    /** @private {number} */
+    this.adIdPointer_ = 0;
+
+    /** @private {?Array} */
+    this.pages_ = null;
   }
 
   /** @override */
   buildCallback() {
-    const ampStory = this.element.parentElement;
-    user().assert(ampStory.tagName === 'AMP-STORY',
+    const ampStoryElement = this.element.parentElement;
+    user().assert(ampStoryElement.tagName === 'AMP-STORY',
         `<${TAG}> should be child of <amp-story>`);
-    this.ampStory_ = ampStory;
-  }
+
+    ampStoryElement.getImpl().then(impl => {
+      this.ampStory_ = impl;
+      this.navigationState_ = this.ampStory_.getNavigationState();
+      this.navigationState_.observe(this.handleStateChange_.bind(this));
+    });
+
+    // TODO(ccordry): move this chunk to layoutCallback
+    // TODO(ccordry): need to fake distance from page to force load
+    for (let i = 0; i < 2; i++) {
+      const mockPage = this.makeMockPage();
+      this.adElements_.push(mockPage);
+      ampStoryElement.appendChild(mockPage);
+    }
+  };
+
 
   /** @override */
   isLayoutSupported() {
     return true;
   }
 
+
   /** @override */
-  layoutCallback() {
-    this.ampStory.appendChild(this.makeMockPage());
-  }
+  // layoutCallback() {}
```

Comment:
remove?

Reply:
I left it as a reminder to myself. Not very elegant, I know 😐

Reply:
now using this

Reply:
this got changed in a later commit